docs/api.rst

```diff
@@ -116,3 +117,83 @@ All sender adaptors are `customization point objects (CPOs)
 .. literalinclude:: ../examples/documentation/when_all_vector_documentation.cpp
    :language: c++
    :start-at: #include
+
+.. _header_pika_cuda:
+
+``pika/cuda.hpp``
+=================
```

Comment:
I agree this might be a good idea. Really this is a symptom of us still calling it `pika/cuda.hpp` etc., but that won't be fixed now... Independently I've toyed with the idea of adding some extra context to the header sections, e.g.

- Runtime initialization etc. (`pika/init.hpp`)
- `std::execution` support (`pika/execution.hpp`)
- CUDA/HIP support (`pika/cuda.hpp`)
- etc.

So I think you've probably at least confirmed that adding some extra context to the headings is useful. I'll think about moving the detailed explanation/note further up as well (though I'm leaning towards keeping it where it is at the moment).

Reply:
I think adding some extra context to the header sections is a good idea. If you end up doing that, then there is no need to move the note further up, since one would see the context instead of just `pika/cuda.hpp`.

Reply:
I've added some more descriptive headings in https://github.com/pika-org/pika/pull/1309/commits/47a0686ecba86d49268aa113ebe0f798f76e0249.

Reply:
LGTM, thanks!